modin/core/storage_formats/pyarrow/query_compiler.py

```diff
@@ -221,6 +220,11 @@ def gandiva_query(table, query):
             expr = gen_table_expr(table, query)
             if not can_be_condition(expr):
                 raise ValueError("Root operation should be a filter.")
+
+            # We use this import here because of https://github.com/modin-project/modin/issues/3849,
+            # after the issue is fixed we should put the import at the top of this file
```

Comment:
Probably add some xfail test?

Reply:
There is no any failing test now.

Reply:
I mean test that checks whether we ready to “put the import…”

Reply:
I see. We could add the following test:
```python
class TestImportGandiva:
    @pytest.mark.xfail(reason="core dump")
    def test_import_gandiva(self):
        import pyarrow.gandiva as gandiva
```
but pytest is not able to handle this segfault. We just need to prioritize #3849 for 0.13 in order not to forget to fix it.

Reply:
We could implement it via launching something like `subprocess.check_call([sys.executable, "-c", "import modin.experimental......PyDbEngine, pyarrow.gandiva"])` which would fail on non-zero exit code

Reply:
@dchigarev will tackle to this.